runtime/queries/metricsview_aggregation.go

```diff
@@ -139,18 +139,28 @@ func (q *MetricsViewAggregation) buildMetricsAggregationSQL(mv *runtimev1.Metric
 
 	selectCols := make([]string, 0, len(q.Dimensions)+len(q.Measures))
 	groupCols := make([]string, 0, len(q.Dimensions))
+	unnestClauses := make([]string, 0)
 	args := []any{}
 
 	for _, d := range q.Dimensions {
 		// Handle regular dimensions
 		if d.TimeGrain == runtimev1.TimeGrain_TIME_GRAIN_UNSPECIFIED {
-			col, err := metricsViewDimensionToSafeColumn(mv, d.Name)
+			dim, err := findDimension(mv, d.Name)
 			if err != nil {
 				return "", nil, err
 			}
-
-			selectCols = append(selectCols, fmt.Sprintf("%s as %s", col, safeName(d.Name)))
-			groupCols = append(groupCols, col)
+			rawColName := getMetricsViewDimensionName(dim)
+			col := safeName(rawColName)
+			unnestColName := safeName(fmt.Sprintf("%s_%s", "unnested", rawColName))
+
+			if dim.Unnest && dialect != drivers.DialectDruid {
+				// select "unnested_colName" as "colName" ... FROM "mv_table", LATERAL UNNEST("mv_table"."colName") tbl("unnested_colName") ...
+				selectCols = append(selectCols, fmt.Sprintf(`%s as %s`, unnestColName, col))
+				unnestClauses = append(unnestClauses, fmt.Sprintf(`, LATERAL UNNEST(%q.%s) tbl(%s)`, mv.Table, col, unnestColName))
```

Comment:
I think we need to use `safeName` on `mv.Table` instead of `%q` (since Go escaping is different from SQL escaping) – but might be wrong, please double check

Reply:
`safeName`  escapes the double quotes using `\` and `%q` should do the same, not sure whats the difference. Also we are using `%q` for escaping table names in comparison toplist sql and possibly other places so will need to fix that as well in case this is wrong.

Reply:
ok changed these and older references as well to `mv.Table` to `safeName(mv.Table)`